sdk/src/service.rs

```diff
@@ -291,3 +302,145 @@ where
         self.router.call(request)
     }
 }
+
+pub struct Config<Plugin> {
+    // Built final plugin
+    pub(crate) plugin: Plugin,
+}
+
+impl Config<()> {
+    pub fn builder() -> config::Builder<IdentityPlugin> {
+        config::Builder {
+            plugin_pipeline: aws_smithy_http_server::plugin::PluginPipeline::default(),
+            aws_auth_authenticate_inserted_at: None,
+            aws_auth_authorize_inserted_at: None,
+            plugin_count: 0,
+        }
+    }
+}
+
+/// Module hosting the builder for `Config`.
+pub mod config {
+    use aws_smithy_http_server::plugin::{IdentityPlugin, PluginPipeline, PluginStack};
+
+    // This follows the same pattern as `MissingOperationsError`.
```

Comment:
I don't think tightness matters here as we're not exposing the innards of these error types for the customer to match on. We also don't plan on matching on them ourselves to handle them. The most we would do is log them; these are essentially unrecoverable errors.

Like, a customer doesn't care what's inside `MissingOperationsError` or `amzn_simple_sdk::simple::config::Error`. We should just make these comfortable to create for ourselves. But since they're trivial to create, I'd just have them hold a `String` so that `.unwrap()` yields something more useful.